File: app/src/main/java/com/madonasyombua/growwithgoogleteamproject/ui/fragment/ProjectsFragment.java
/*Copyright (c) 2018 Madona Syombua

        Licensed under the Apache License, Version 2.0 (the "License");
        you may not use this file except in compliance with the License.
        You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

        Unless required by applicable law or agreed to in writing, software
        distributed under the License is distributed on an "AS IS" BASIS,
        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
        See the License for the specific language governing permissions and
        limitations under the License.
 */

package com.madonasyombua.growwithgoogleteamproject.ui.fragment;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.madonasyombua.growwithgoogleteamproject.R;
import com.madonasyombua.growwithgoogleteamproject.ui.adapter.PortfolioAdapter;
import com.madonasyombua.growwithgoogleteamproject.interfaces.OnFragmentInteractionListener;
import com.madonasyombua.growwithgoogleteamproject.data.models.Portfolio;

import java.util.ArrayList;
import java.util.List;

import static com.facebook.FacebookSdk.getApplicationContext;

/**
 * A simple {@link Fragment} subclass.
 * Activities that contain this fragment must implement the
 *
 * to handle interaction events.
 * Use the {@link ProjectsFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class ProjectsFragment extends Fragment {
    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";
    private final List<Portfolio> portfolioList = new ArrayList<>();
    private PortfolioAdapter mAdapter;

    private OnFragmentInteractionListener mListener;

    public ProjectsFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment ProjectsFragment.
     */
    // TODO: Rename and change types and number of parameters
    public static ProjectsFragment newInstance(String param1, String param2) {
        ProjectsFragment fragment = new ProjectsFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            // TODO: Rename and change types of parameters
            String mParam1 = getArguments().getString(ARG_PARAM1);
            String mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_projects, container, false);
        RecyclerView recyclerView = view.findViewById(R.id.portfolio_recycler_view);

        mAdapter = new PortfolioAdapter(portfolioList);
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(mLayoutManager);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setAdapter(mAdapter);

        testPortfolioData();


        return view;
    }

    // TODO: Rename method, update argument and hook method into UI event
    public void onButtonPressed(Uri uri) {
        if (mListener != null) {
            mListener.onFragmentInteraction(uri);
        }
    }

    private void testPortfolioData() {
        Portfolio portfolio = new Portfolio("Simple Maths", "Android Application", "Simple Maths is a" +
                "mobile app project that I did for kids. It's a learning app for Arithmetic", R.drawable.ic_facebook);
        portfolioList.add(portfolio);

        portfolio = new Portfolio("Essay Tutors", "Web Development", "This is an online tutoring network" +
                "for teachers and students.", R.drawable.logo);
        portfolioList.add(portfolio);

        portfolio = new Portfolio("Budgeting Buddy", "Android Development", "This is a budgeting planner" +
                " and expense tracking mobile device for Android devices.", R.drawable.ic_google);
        portfolioList.add(portfolio);

        mAdapter.notifyDataSetChanged();
    }

   /* @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }

        ((AppCompatActivity)(context)).getSupportActionBar().setTitle(getString(R.string.projects));
    }*/

       // ((AppCompatActivity)(context)).getSupportActionBar().setTitle(getString(R.string.projects));


    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }
}

File: app/src/main/java/com/madonasyombua/growwithgoogleteamproject/ui/fragment/ProjectsFragment.kt
/*Copyright (c) 2018 Madona Syombua

        Licensed under the Apache License, Version 2.0 (the "License");
        you may not use this file except in compliance with the License.
        You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

        Unless required by applicable law or agreed to in writing, software
        distributed under the License is distributed on an "AS IS" BASIS,
        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
        See the License for the specific language governing permissions and
        limitations under the License.
 */
package com.madonasyombua.growwithgoogleteamproject.ui.fragment

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.facebook.FacebookSdk
import com.madonasyombua.growwithgoogleteamproject.R
import com.madonasyombua.growwithgoogleteamproject.data.models.Portfolio
import com.madonasyombua.growwithgoogleteamproject.databinding.FragmentProjectsBinding
import com.madonasyombua.growwithgoogleteamproject.interfaces.OnFragmentInteractionListener
import com.madonasyombua.growwithgoogleteamproject.ui.adapter.PortfolioAdapter
import java.util.*

class ProjectsFragment : Fragment(R.layout.fragment_projects) {
    private val portfolioList: MutableList<Portfolio> = ArrayList()
    private var mAdapter: PortfolioAdapter? = null
    private var mListener: OnFragmentInteractionListener? = null
    private var fragmentProjectsBinding: FragmentProjectsBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            val mParam1 = arguments?.getString(ARG_PARAM1)
            val mParam2 = arguments?.getString(ARG_PARAM2)
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val binding = FragmentProjectsBinding.bind(view)
        fragmentProjectsBinding = binding

        mAdapter = PortfolioAdapter(portfolioList)
        val mLayoutManager: RecyclerView.LayoutManager =
            LinearLayoutManager(FacebookSdk.getApplicationContext())
        binding.portfolioRecyclerView.layoutManager = mLayoutManager
        binding.portfolioRecyclerView.itemAnimator = DefaultItemAnimator()
        binding.portfolioRecyclerView.adapter = mAdapter
        testPortfolioData()

    }

    private fun testPortfolioData() {
        var portfolio = Portfolio(
            "Simple Maths",
            "Android Application",
            "Simple Maths is a" +
                    "mobile app project that I did for kids. It's a learning app for Arithmetic",
            R.drawable.ic_facebook
        )
        portfolioList.add(portfolio)
        portfolio = Portfolio(
            "Essay Tutors", "Web Development", "This is an online tutoring network" +
                    "for teachers and students.", R.drawable.logo
        )
        portfolioList.add(portfolio)
        portfolio = Portfolio(
            "Budgeting Buddy", "Android Development", "This is a budgeting planner" +
                    " and expense tracking mobile device for Android devices.", R.drawable.ic_google
        )
        portfolioList.add(portfolio)
        mAdapter?.notifyDataSetChanged()
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    companion object {

        private const val ARG_PARAM1 = "param1"
        private const val ARG_PARAM2 = "param2"

    }

    override fun onDestroyView() {
        fragmentProjectsBinding = null
        super.onDestroyView()
    }

}